sklearn/model_selection/_split.py

```diff
@@ -637,6 +637,121 @@ def split(self, X, y, labels=None):
         """
         return super(StratifiedKFold, self).split(X, y, labels)
 
+class HomogeneousTimeSeriesCV(_BaseKFold):
+    """Homogeneous Time Series cross-validator
+
+    Provides train/test indices to split time series data in train/test sets.
+
+    This cross-validation object is a variation of KFold.
+    In iteration k, it returns first k folds as train set and k+1 fold as
+    test set.
+
+    Read more in the :ref:`User Guide <cross_validation>`.
+
+    Parameters
+    ----------
+    n_folds : int, default=3
+        Number of folds. Must be at least 2.
+
+    Examples
+    --------
+    >>> from sklearn.model_selection import HomogeneousTimeSeriesCV
+    >>> X = np.array([[1, 2], [3, 4], [1, 2], [3, 4]])
+    >>> y = np.array([1, 2, 3, 4])
+    >>> htscv = HomogeneousTimeSeriesCV(n_folds=4)
```

Comment:
If we split data into `n_folds` folds,
then there are actually only `n_folds-1` number of splits for `TimeSeriesCV` since the first fold can never be a test fold.